ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -219,16 +219,15 @@ message DisclosedContract {
   // Required
   string contract_id = 2;
 
-  // The contract arguments
+  // The contract arguments as typed Record
   // Required
-  oneof arguments {
-    // The contract arguments as typed Record
-    Record create_arguments = 3;
+  Record create_arguments = 3;
 
-    // The contract arguments specified using an opaque blob extracted from the ``create_arguments_blob`` field
-    // of a ``com.daml.ledger.api.v1.CreatedEvent``.
-    google.protobuf.Any create_arguments_blob = 5;
-  }
+  // The contract arguments specified using an opaque blob extracted from the ``create_arguments_blob`` field
+  // of a ``com.daml.ledger.api.v1.CreatedEvent``. Ledger API server validates this data in conjunction with
+  // the ``create_arguments`` record and returns a comprehensive error when any inconsistency is found.
+  // Required
```

Comment:
Why is this required?

Can't I use either of the two, and if I use both get additional validation applied?

Reply:
Why do we even need this change away from the `oneof`?

Reply:
> Make the blob mandatory on the command submission

This is expensive for consumers like scribe (aka PQS): they now have to always store both the expanded form of the create arguments (for indexing) and the full blob for all templates they subscribe to. Thereby requiring double the amount of storage.

What's the rationale for this change?

Reply:
Only the blob will be mandatory and is a consequence of the upgrading story. The client will likely use a code-generated stub to handle the contracts which may be based on an older version that will drop some of the contract fields

Reply:
OK. I can see that upgrading might require always passing `create_argument_blob`s for disclosed contracts.

Do you have a design doc clarifying the exact requirement(s) and why this solution solves it, as well as the limitations (like the 2x storage increase for PQS for contracts that need to be disclosed?).

Reply:
I have subsequently changed the PR to allow for backwards compatibility. Again my change was stuck due to repo being borked

Reply:
We have an extension to the original ED document:
https://docs.google.com/document/d/1uhDEAKLOqVHXiNOUNcNtABlAzvo1ELRdnDEXOWVIQyg/edit#bookmark=id.2pwvqe594wr8

Reply:
Thanks for the clarifications @mziolekda . It looks like we're converging here. I've added a comment on the design doc.

Reply:
I don't think our code-gen types should drop unknown fields. Can we preserve them (without interpretation), and include them in any output format, similar to the protobuf behaviour?